src/partitioner/local/RDFParser.cpp

```diff
@@ -0,0 +1,422 @@
+    try {
+        XMLPlatformUtils::Initialize();  // Initialize Xerces infrastructure
+    }
+    catch (XMLException &e) {
+        char *message = XMLString::transcode(e.getMessage());
+        cerr << "XML toolkit initialization error: " << message << endl;
+        XMLString::release(&message);
+    }
+
+    // Tags and attributes used in XML file.
+    // Can't call transcode till after Xerces Initialize()
+    TAG_root = XMLString::transcode("root");
+    TAG_RDF = XMLString::transcode("rdf:RDF");
+    TAG_has_author = XMLString::transcode("akt:has-author");
+
+    TAG_has_title = XMLString::transcode("akt:has-title");
+    TAG_article_of_journal = XMLString::transcode("akt:article-of-journal");
+    TAG_Journal = XMLString::transcode("akt:Journal");
+    TAG_has_volume = XMLString::transcode("akt:has-volume");
+    TAG_has_web_address = XMLString::transcode("akt:has-web-address");
+    TAG_Calendar_date = XMLString::transcode("akts:Calendar-Date");
+    TAG_year_of = XMLString::transcode("akts:year-of");
+    TAG_month_of = XMLString::transcode("akts:month-of");
+    TAG_has_date = XMLString::transcode("akt:has-date");
+
+
+    TAG_Article_Reference = XMLString::transcode("akt:Article-Reference");
+    TAG_Book_Section_Reference = XMLString::transcode("akt:Book-Section-Reference");
+    TAG_Thesis_Reference = XMLString::transcode("akt:Thesis-Reference");
+    TAG_Book_Reference = XMLString::transcode("akt:Book-Reference");
+    TAG_Conference_Proceedings_Reference = XMLString::transcode("akt:Conference-Proceedings-Reference");
+
+    TAG_Person = XMLString::transcode("akt:Person");
+    TAG_full_name = XMLString::transcode("akt:full-name");
+
+
+    m_ConfigFileParser = new XercesDOMParser;
+}
+
+/**
+ *  Class destructor frees memory used to hold the XML tag and
+ *  attribute definitions. It als terminates use of the xerces-C
+ *  framework.
+ */
+
+GetConfig::~GetConfig() {
+    // Free memory
+
+    delete m_ConfigFileParser;
+
+    try {
+        XMLString::release(&TAG_root);
+
+        XMLString::release(&TAG_RDF);
+        XMLString::release(&TAG_has_author);
+        XMLString::release(&TAG_has_title);
+        XMLString::release(&TAG_article_of_journal);
+        XMLString::release(&TAG_Journal);
+        XMLString::release(&TAG_has_volume);
+        XMLString::release(&TAG_has_web_address);
+        XMLString::release(&TAG_Calendar_date);
+        XMLString::release(&TAG_year_of);
+        XMLString::release(&TAG_month_of);
+        XMLString::release(&TAG_has_date);
+        XMLString::release(&TAG_Article_Reference);
+        XMLString::release(&TAG_Book_Section_Reference);
+        XMLString::release(&TAG_Thesis_Reference);
+        XMLString::release(&TAG_Book_Reference);
+        XMLString::release(&TAG_Conference_Proceedings_Reference);
+        XMLString::release(&TAG_Person);
+        XMLString::release(&TAG_full_name);
+    }
+    catch (...) {
+        cerr << "Unknown exception encountered in TagNamesdtor" << endl;
```

Comment:
What is TagNamesdtor?

Reply:
It was from an example code I was referring. we need to define the TAGs which are in XML file. Here it throws an error if I try to release unknown TAG. TAGNamesdtor could be TAG name director

Reply:
I have seen the same error message in multiple other places. The use of TagNamesdtor seems no meaning. Instead can you change the error message to "Unknown exception encountered while trying to release unknown TAG"? That error message is more meaningful, rather than having in the error message "TagNamesdtor" which we do not know the meaning.